components/event-receiver/event-input-adapters/org.wso2.carbon.event.input.adapter.sqs/src/main/java/org/wso2/carbon/event/input/adapter/sqs/SQSEventAdapter.java

```diff
@@ -0,0 +1,175 @@
+                    .VISIBILITY_TIMEOUT)));
+        }
+
+        if (adapterProperties.get(SQSEventAdapterConstants.DELETE_AFTER_CONSUMING) != null) {
+            sqsConfig.setShouldDeleteAfterConsuming(Boolean.parseBoolean(adapterProperties
+                    .get(SQSEventAdapterConstants.DELETE_AFTER_CONSUMING)));
+        }
+
+        if (adapterProperties.get(SQSEventAdapterConstants.RETRY_INTERVAL) != null) {
+            sqsConfig.setRetryInterval(Integer.parseInt(adapterProperties
+                    .get(SQSEventAdapterConstants.RETRY_INTERVAL)));
+        }
+
+        if (adapterProperties.get(SQSEventAdapterConstants.RETRY_ATTEMPT_LIMIT) != null) {
+            sqsConfig.setRetryCountLimit(Integer.parseInt(adapterProperties
+                    .get(SQSEventAdapterConstants.RETRY_ATTEMPT_LIMIT)));
+        }
+
+        if (globalProperties.get(SQSEventAdapterConstants.ADAPTER_LISTENER_WAITING_TIME) != null) {
+            listenerWaitingTime = Integer.parseInt(globalProperties
+                    .get(SQSEventAdapterConstants.ADAPTER_LISTENER_WAITING_TIME));
+        }
+
+        if (globalProperties.get(SQSEventAdapterConstants.NUMBER_OF_THREADS) != null) {
+            numberOfThreads = Integer.parseInt(globalProperties
+                    .get(SQSEventAdapterConstants.NUMBER_OF_THREADS));
+        }
+
+        scheduler = Executors.newScheduledThreadPool(numberOfThreads);
+
+        int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+        sqsProvider = new SQSProvider(sqsConfig, eventAdaptorListener, tenantId);
+    }
+
+    @Override
+    public void testConnect() throws TestConnectionNotSupportedException {
+        throw new TestConnectionNotSupportedException("not-supported");
+    }
+
+    @Override
+    public void connect() {
+        if (scheduler != null && !scheduler.isShutdown()) {
+            for (int i = 0; i < numberOfThreads; i++) {
+                scheduler.scheduleAtFixedRate(sqsProvider.getNewSQSTask(), listenerWaitingTime, pollingInterval,
+                        TimeUnit.MILLISECONDS);
+            }
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        if (scheduler != null) {
+            scheduler.shutdown();
+        }
+    }
+
+    @Override
+    public void destroy() {
+        if (scheduler != null) {
+            scheduler.shutdown();
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof SQSEventAdapter)) {
+            return false;
+        }
+        SQSEventAdapter that = (SQSEventAdapter) o;
+        return id.equals(that.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return id.hashCode();
+    }
+
+    @Override
+    public boolean isEventDuplicatedInCluster() {
+        return false;
+    }
+
+    @Override
+    public boolean isPolling() {
+        return false;
```

Comment:
How come this is not a polling receiver? This polls messages from a queue right (ref:SQSTask.java)?